src/main/java/reposense/git/GitLsTree.java

```diff
@@ -19,14 +19,18 @@
  * Git ls tree is responsible to obtain the list of staged files of a branch.
  */
 public class GitLsTree {
+    // Although forward-slash (/) is an invalid character in Windows file path, it is not included in the regex as
+    // for the output of git-ls-tree, files in directories are separated by forward-slash (e.g.: folder/name/file.txt).
+    // Also, it is not possible to create and commit files with forward-slash characters in UNIX OSes.
+    private static final Pattern ILLEGAL_WINDOWS_FILE_PATTERN = Pattern.compile(
+            "^((CON|PRN|AUX|NUL|COM[1-9]|LPT[1-9])(?=\\..*|$))" // contains reserved values (e.g.: com1)
+                    + "|[<>:\"\\\\|?*\\x00-\\x1F]"  // contains any reserved characters in directory name
+                    + "|(^\\s)|(/\\s+)"             // file or folder names with leading whitespaces
+                    + "|([\\s.]$)|([\\s.]+/)",      // folder or file names ending with period or whitespaces
+            Pattern.CASE_INSENSITIVE);
```

Comment:
MSDN Documentation on windows file/directory naming:
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file?redirectedfrom=MSDN

The regex used is extracted from the one given [here](https://stackoverflow.com/a/6804755/11570538), but modified such that: -
- The logic is reversed to capture illegal/reserved characters/names in the directories and to output to the user the offending character/name
- the regex adapts to the file names output by git-ls-tree command.

On the side note, I decided to put comments beside every lines here due to the complexity of this regex, though not sure if it may be excessive or following the convention.